website/templates/project/view_file.mako

```diff
@@ -8,7 +8,20 @@
 <div class="row">
   <div class="col-sm-5">
     <h2 class="break-word">
-      ${file_name | h}
+      ## Split file name into two parts: with and without extension
+      <%
+       import os
+       part1, part2 = os.path.splitext(file_name)
+       ## POSIX standard for chars in a fully portable filename
+       validFileExtension = all(i.isalnum() or i in ['.', '_', '-'] for i in part2)
+       if validFileExtension:
```

Comment:
When a file name doesn't have an extension, but the file name contains a ".", like for "Really long title 10.26.15 important stuff", the "15 important stuff" was treated as an actual extension when it wasn't supposed to be.
The code in line 16 checks if the result from os.path.splitext() is really supposed to be an extension.